activitychart-basic.html

```diff
@@ -0,0 +1,274 @@
+          .data(data)
+          .enter().append('rect')
+          .attr('width', this.squareWidth)
+          .attr('height', this.squareWidth)
+          .attr('x', this._spaceSquareHor)
+          .attr('y', this._spaceSquareVert)
+          .attr('fill', function(d) { return colorScale(d.y); })
+          .on('mouseover', this._showLabel)
+          .on('mouseout', this._hideLabel);
+      },
+
+      _showLabel: function (d, i) {
+        //using a group element 'label' with an appended text element and a rect element as background to display a label
+
+        var polyElement = Polymer.dom(this).parentNode.parentNode.parentNode;
+        var label = d3.select('svg').append('g')
+          .attr('id', 'label-' + i);
+
+        //append rect element first so that following text element is displayed above rect
+        label.append('rect');
+        var labelText = label.append('text')
+					.text(d.y + ' at ' + d.x)
+					.style('font-size', '0.8em')
+					.attr('fill', 'black');
+
+        var labelWidth = labelText.node().getBBox().width;
+        var labelHeight = labelText.node().getBBox().height;
+
+        //this split is necessary because the position is set relative to the size of the text element
+        labelText
+          .attr('x', polyElement._spaceLabelHor(d, i, labelWidth))
+          .attr('y', polyElement._spaceLabelVert(d, i, labelHeight));
+
+        //finish rect configuration that was appended previously
+        label.select('rect')
+          .attr('x', labelText.attr('x') - 1)
+  				.attr('y', labelText.attr('y') - 0.7 * labelHeight) // 0.7 as factor because labelHeight is bigger than actual height of text
+  				.attr('fill', 'white')
+  				.attr('width', labelWidth + 1) //ok, so that it doesn't become too short for the text because of the visual offset above
+      		.attr('height', 0.8 * labelHeight);
+			},
+
+      _hideLabel: function (d, i) {
+        d3.select('#label-' + i).remove();
+      },
+
+      _spaceLabelHor: function(d, i, labelWidth) {
+        var x = this._spaceSquareHor(d, i);
+        //calculate the maximum x position possible so that the label won't be cut off on the edge of the SVG
+        var maxPosition = d3.select('svg').attr('width') - (this.squareSpacing - this.squareWidth) - labelWidth;
+
+        if(x > maxPosition) { // label would normally be "shown" outside the SVG space
+          x = maxPosition;
+        }
+        return x;
+      },
+
+      _spaceLabelVert: function(d, i, labelHeight) {
+        var y = this._spaceSquareVert(d, i) - 2;
+        if(y < 0) { // label would normally be "shown" just exactly outside the SVG space
+          y = this.squareWidth + 0.7 * labelHeight;
+        }
+        return y;
+      },
+
+      _spaceSquareHor: function (d, i) {
+        var polyElement = this;
+        if (Polymer.dom(this).node.nodeName != "ACTIVITYCHART-BASIC") {
+          var polyElement = Polymer.dom(this).parentNode.parentNode.parentNode;
```

Comment:
see above, there got to be a better way to do this than `.parentNode.parentNode.parentNode`